src/test/java/org/broadinstitute/hellbender/tools/PrintReadsIntegrationTest.java

```diff
@@ -39,4 +50,59 @@ public void testNoConflictPG() throws IOException {
         Assert.assertNotNull(SamReaderFactory.makeDefault().open(outFile).getFileHeader().getProgramRecord("GATK PrintReads"));
         Assert.assertNotNull(SamReaderFactory.makeDefault().open(outFile).getFileHeader().getProgramRecord("GATK PrintReads.1"));
     }
+
+    @DataProvider
+    public Object[][] getHttpPaths(){
+        final String bam = "gs://hellbender/test/resources/benchmark/CEUTrio.HiSeq.WEx.b37.NA12892.bam";
+        final String bai = "gs://hellbender/test/resources/benchmark/CEUTrio.HiSeq.WEx.b37.NA12892.bam.bai";
+        final String cram = "gs://hellbender/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.cram";
+        final String crai = "gs://hellbender/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.cram.bai";
+        final List<SimpleInterval> largeFileIntervals = Arrays.asList(new SimpleInterval("3",1_000_000, 1_000_001),
+               new SimpleInterval("3", 1_000_003, 1_000_100),
+                new SimpleInterval("20", 1_099_000, 1_100_000));
+
+        final List<SimpleInterval> cramIntervals = Arrays.asList(new SimpleInterval("20", 9_999_902, 10_000_000));
+        return new Object[][]{
+                {BucketUtils.bucketPathToPublicHttpUrl(bam), BucketUtils.bucketPathToPublicHttpUrl(bai), bam, bai, largeFileIntervals, 528L},
+                {BucketUtils.createSignedUrlToGcsObject(bam, 1L), BucketUtils.createSignedUrlToGcsObject(bai, 1L), bam, bai, largeFileIntervals, 528L},
+                {BucketUtils.bucketPathToPublicHttpUrl(cram), BucketUtils.bucketPathToPublicHttpUrl(crai), cram, crai, cramIntervals, 112L},
+                {BucketUtils.createSignedUrlToGcsObject(cram, 1L), BucketUtils.createSignedUrlToGcsObject(crai, 1L), cram, crai, cramIntervals, 112L}
```

Comment:
I assume that you've verified these expected read counts using a code path that does not go through the new HTTP NIO provider?

Reply:
I used the standard GCS nio provider as well.   I looked manually, but the test gets it with both http and gcs and compares the results to be sure they are the same.